### PR TITLE
switch to using interface func for consistency

### PR DESCRIPTION
**Pull request by luthermonson**
**What type of PR is this?**
Not sure why the interface func wasn't on this line when it was used a few lines below. updating for consistency.
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by sedefsavas** [comment]
/hold
until https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3720 is merged.
All upcoming PRs need to be merged once v1beta2 API types are in.

File: pkg/cloud/services/eks/identity_provider.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 
 func (s *Service) reconcileIdentityProvider(ctx context.Context) error {
 	s.scope.Info("reconciling oidc identity provider")
-	if s.scope.ControlPlane.Spec.OIDCIdentityProviderConfig == nil {
+	if s.scope.OIDCIdentityProviderConfig() == nil {
 		return nil
```
Comment (shivi28) [inline pkg/cloud/services/eks/identity_provider.go:35]: Nit: we can add some info log before returning from here.

**Comment by shivi28** [comment]
Thanks @luthermonson for the PR. Added a nit comment, else /lgtm

File: pkg/cloud/services/eks/identity_provider.go (hunk 1 above)
Comment (luthermonson) [reply pkg/cloud/services/eks/identity_provider.go:36]: fixed

**Comment by luthermonson** [comment]
@shivi28 ready

**Comment by luthermonson** [comment]
@shivi28 can this be merged?

**Comment by Skarlso** [comment]
>/lgtm
>/approve

@shivi28 Huh, oddly, this didn't work. :O

**Pull request merged**